lib/ansible/plugins/connection/ssh.py

```diff
@@ -627,6 +655,8 @@ def __init__(self, *args: t.Any, **kwargs: t.Any) -> None:
             self.module_implementation_preferences = ('.ps1', '.exe', '')
             self.allow_executable = False
 
+        self._populated_agent: str | None = None
```

Comment:
can we use pathlib objects for new things plz?

Reply:
Done.